test/uk/gov/hmrc/transitmovements/services/ObjectStoreServiceSpec.scala

```diff
@@ -0,0 +1,110 @@
+ */
+
+package uk.gov.hmrc.transitmovements.services
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.reset
+import org.mockito.MockitoSugar.when
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.ScalaFutures.whenReady
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.objectstore.client.Md5Hash
+import uk.gov.hmrc.objectstore.client.Object
+import uk.gov.hmrc.objectstore.client.ObjectMetadata
+import uk.gov.hmrc.objectstore.client.Path
+import uk.gov.hmrc.objectstore.client.Path.File
+import uk.gov.hmrc.objectstore.client.play.PlayObjectStoreClient
+import uk.gov.hmrc.transitmovements.base.StreamTestHelpers
+import uk.gov.hmrc.transitmovements.base.TestActorSystem
+import uk.gov.hmrc.transitmovements.generators.ModelGenerators
+import uk.gov.hmrc.transitmovements.models.ObjectStoreURI
+import uk.gov.hmrc.transitmovements.services.errors.ObjectStoreError
+
+import java.time.Instant
+import java.util.UUID.randomUUID
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ObjectStoreServiceSpec
+    extends AnyFreeSpec
+    with Matchers
+    with MockitoSugar
+    with TestActorSystem
+    with StreamTestHelpers
+    with BeforeAndAfterEach
+    with ModelGenerators
+    with ScalaCheckDrivenPropertyChecks {
+
+  implicit val hc: HeaderCarrier                           = HeaderCarrier()
+  private val mockObjectStoreClient: PlayObjectStoreClient = mock[PlayObjectStoreClient]
+
+  override def beforeEach: Unit =
+    reset(mockObjectStoreClient)
+
+  "Object Store service" - {
+
+    "should return the contents of a file" in {
+      val filePath =
+        Path.Directory(s"common-transit-convention-traders/movements/${arbitraryMovementId.arbitrary.sample.get}").file(randomUUID.toString).asUri
+      val metadata    = ObjectMetadata("", 0, Md5Hash(""), Instant.now(), Map.empty[String, String])
+      val content     = "content"
+      val fileContent = Option[Object[Source[ByteString, NotUsed]]](Object.apply(File(filePath), Source.single(ByteString(content)), metadata))
+      when(mockObjectStoreClient.getObject[Source[ByteString, NotUsed]](any[File](), any())(any(), any())).thenReturn(Future.successful(fileContent))
+      val service = new ObjectStoreServiceImpl(mockObjectStoreClient)
+      val result  = service.getObjectStoreFile(ObjectStoreURI(filePath))
+      whenReady(result.value) {
+        r =>
+          r.isRight mustBe true
+          r.toOption.get mustBe fileContent.get.content
```

Comment:
Can this be shortened to `r mustBe Right(content)`?

Reply:
I tried but I am getting  Right(Source(SourceShape(single.out(1854403227)))) was not equal to Right("content") . So I used r.toOption.get mustBe fileContent.get.content